src/esn.calendar.libs/app/services/calendar-api.js

```diff
@@ -316,5 +317,21 @@ require('./http-response-handler.js');
       return calDavRequest('get', calendarPath, { Accept: CAL_ACCEPT_EXPORT_HEADER })
         .then(calHttpResponseHandler(200, _.property('data')));
     }
+    /**
+      * POST resuest used to generate token for secret link to download ics calendar
+      * @param {Object} jwtPayload   paylod to generate token
+      */
+    function generateToken(jwtPayload) {
+
+      return calendarRestangular.one('calendars/generateJWTforSecretLink').customPOST(jwtPayload)
+        .then(calHttpResponseHandler(200, _.property(['data'])))
```

Comment:
```suggestion
        .then(calHttpResponseHandler(200, _.property('data')))
```
does this work without the `[ ]`?

Reply:
yes , it is working , in the first i want to fetch a nested object forthat i used an array  like this  _.property(['data', 'token'] but got undefined